WalletWasabi.Fluent/ViewModels/SearchBar/Sources/ActionsSource.cs

```diff
@@ -0,0 +1,59 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Reactive.Linq;
+using System.Threading.Tasks;
+using DynamicData;
+using WalletWasabi.Fluent.ViewModels.Navigation;
+using WalletWasabi.Fluent.ViewModels.SearchBar.Patterns;
+using WalletWasabi.Fluent.ViewModels.SearchBar.SearchItems;
+
+namespace WalletWasabi.Fluent.ViewModels.SearchBar.Sources;
+
+public class ActionsSource : ISearchItemSource
+{
+	public IObservable<IChangeSet<ISearchItem, ComposedKey>> Changes => GetItemsFromMetadata()
+		.ToObservable()
+		.ToObservableChangeSet(x => x.Key);
+
+	private static IEnumerable<ISearchItem> GetItemsFromMetadata()
+	{
+		return NavigationManager.MetaData
+			.Where(m => m.Searchable)
+			.Select(m =>
+			{
+				var func = CreateFunc(m);
+				var searchItem = new ActionableItem(m.Title, m.Caption, func, m.Category ?? "No category", m.Keywords)
+				{
+					Icon = m.IconName,
+					IsDefault = true,
+				};
+				return searchItem;
+			});
+	}
+
+	private static Func<Task> CreateFunc(NavigationMetaData navigationMetaData)
+	{
+		return async () =>
+		{
+			var vm = await NavigationManager.MaterialiseViewModelAsync(navigationMetaData);
+			if (vm is null)
+			{
+				return;
+			}
+
+			Navigate(vm.DefaultTarget).To(vm);
+		};
+	}
+
+	private static INavigationStack<RoutableViewModel> Navigate(NavigationTarget currentTarget)
+	{
+		return currentTarget switch
+		{
+			NavigationTarget.HomeScreen => NavigationState.Instance.HomeScreenNavigation,
+			NavigationTarget.DialogScreen => NavigationState.Instance.DialogScreenNavigation,
+			NavigationTarget.FullScreen => NavigationState.Instance.FullScreenNavigation,
+			NavigationTarget.CompactDialogScreen => NavigationState.Instance.CompactDialogScreenNavigation,
+			_ => throw new NotSupportedException()
+		};
+	}
```

Comment:
Remove code duplication.

```suggestion
			RoutableViewModel.Navigate(vm.DefaultTarget).To(vm);
		};
	}
```